src/app/providers/profile/profile.ts

```diff
@@ -1799,6 +1878,20 @@ export class ProfileProvider {
     return this.wallet[walletId];
   }
 
+  public getWalletPrimary(walletId, tokenId) {
+    let wallet = _.cloneDeep(this.wallet[walletId]);
+    if (wallet && !tokenId) {
+      return wallet;
+    } else if (wallet && walletId) {
```

Comment:
condition not clear => we should check wallet && wallet.tokens ? i don't see walletId important in this condition